lib/shopify/resources/metafield.ex

```diff
@@ -15,4 +32,53 @@ defmodule Shopify.Metafield do
     :value_type,
     :updated_at
   ]
+
+  @doc false
+  def empty_resource do
+    %Metafield{
+      key: "",
+      value: "",
+      value_type: "string",
+      namespace: ""
+    }
+  end
+
+  [:blogs, :collections, :customers, :draft_orders, :orders, :pages, :products]
+  |> Enum.each(fn (name) ->
+    def unquote(name)(session, parent_id, params \\ %{}) do
+      session
+      |> Request.new(metafield_url(Atom.to_string(unquote(name)), parent_id), params, plural_resource())
```

Comment:
More of a code style suggestion than an issue here, but I think if we have to cast to string anyways, then using strings instead of atoms makes it slightly more readable and we could also use the `~w` sigil to reduce noise.

We could do something like this:

```elixir
~w(blogs collections customers draft_orders orders pages products)
|> Enum.each(fn (name) ->
  def unquote(:"#{name}")(session, parent_id, params \\ %{}) do
    url = metafield_url(unquote(name), parent_id)
    session
    |> Request.new(url, params, plural_resource())
  end
end)
```

take it with a grain of salt, I haven't actually tested this 🌶